packages/medusa/src/api/routes/admin/products/create-product.ts

```diff
@@ -132,29 +132,39 @@ export default async (req, res) => {
   const entityManager: EntityManager = req.scope.resolve("manager")
   const productModuleService = req.scope.resolve("productModuleService")
 
-  if (productModuleService) {
-    const createProductWorkflow = createProducts(req.scope)
-
-    const input = {
-      products: [
-        validated,
-      ] as WorkflowTypes.ProductWorkflow.CreateProductInputDTO[],
-      config: {
-        listConfig: {
-          select: defaultAdminProductFields,
-          relations: defaultAdminProductRelations,
+  const isWorkflowEnabled = featureFlagRouter.isFeatureEnabled({
+    workflows: Workflows.CreateProducts,
+  })
+
+  if (isWorkflowEnabled) {
+    if (productModuleService) {
+      const createProductWorkflow = createProducts(req.scope)
+
+      const input = {
+        products: [
+          validated,
+        ] as WorkflowTypes.ProductWorkflow.CreateProductInputDTO[],
+        config: {
+          listConfig: {
+            select: defaultAdminProductFields,
+            relations: defaultAdminProductRelations,
+          },
         },
-      },
-    }
+      }
 
-    const { result: products } = await createProductWorkflow.run({
-      input,
-      context: {
-        manager: entityManager,
-      },
-    })
+      const { result: products } = await createProductWorkflow.run({
+        input,
+        context: {
+          manager: entityManager,
+        },
+      })
 
-    return res.json({ product: products[0] })
+      return res.json({ product: products[0] })
+    } else {
+      logger.warn(
+        `Cannot run ${Workflows.CreateProducts} workflow without '@medusajs/product' installed`
+      )
```

Comment:
The class WorkflowManager has 2 properties that are not being used at this moment. `requiredModules?: Set<string>` and `optionalModules?: Set<string>`
We can think about add this kind of validation / warning at the workflow manager level.
In my opinion we should throw in this case. The module is required, and FF on.

Reply:
Yes once it will be used and the end point will be replaced by the workflow only, we will remove the check on product, but for now we are in the middle step, both workflow and end point exists, so if the module is required and not present we should still complete using the current end point flow instead of not doing the action i think. Wdyt?